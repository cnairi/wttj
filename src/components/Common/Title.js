import React from 'react'
import { string } from 'prop-types'
import { Box } from '@welcome-ui/box'

import * as S from './styles'

function Title({ text }) {
  return (
    <Box margin="1.875rem 0" position="relative">
      <S.SectionTitle variant="h2">{text}</S.SectionTitle>
    </Box>
  )
}

Title.propTypes /* remove-proptypes */ = {
  text: string,
}

export default Title
