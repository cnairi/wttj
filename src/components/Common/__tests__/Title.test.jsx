import React from 'react'
import Title from '@components/Common/Title'
import { render } from '@utils/tests'

const text = 'Media'

describe('Component Common <Title />', () => {
  test('It should renders properly', () => {
    const tree = render(<Title text={text} />)
    expect(tree).toMatchSnapshot()
  })

  test('It should display a title', async () => {
    const { getByRole } = render(<Title text={text} />)
    const titleElement = getByRole('heading')
    expect(titleElement).toHaveTextContent(text)
  })
})
