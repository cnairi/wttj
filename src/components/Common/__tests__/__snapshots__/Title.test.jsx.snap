// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Common <Title /> It should renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0.c0 {
  margin: 1.875rem 0;
  position: relative;
}

.c1 {
  font-family: welcome-font,sans-serif;
  font-weight: 600;
  font-size: 2.25rem;
  line-height: 1.2;
  display: block;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 35px;
  padding-bottom: 0.625rem;
  border-bottom: 1px solid rgb(229,229,229);
}

.c2::before {
  content: '';
  position: absolute;
  left: 0px;
  height: 4px;
  width: 20px;
  background-color: rgb(255,205,0);
  top: 0.875rem;
}

<body>
    <div>
      <div
        class="c0"
      >
        <h2
          class="c1 c2 wui-text"
        >
          Media
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKC sc-LzLsE wenIp"
    >
      <h2
        class="sc-LzLsF fRbamu sc-LzLte bklnsS wui-text"
      >
        Media
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
